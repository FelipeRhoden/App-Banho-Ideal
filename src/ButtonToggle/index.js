import React, { useState } from 'react';
import { View, Image, TouchableOpacity } from 'react-native';

export default class ButtonToggle extends React.Component {

    constructor(props){
        props = props || {};
        props.trigger1 = props.trigger1 || (()=>{});
        props.trigger2 = props.trigger2 || (()=>{});
        props.picture = props.picture || '';
        super(props);

        this.state = {
            toggle : true,
        }
    }

    toggle(){
        this.setState({toggle : !this.state.toggle});
    }

    render(){
        return(
            <View>
                {this.state.toggle
                ?
                <TouchableOpacity onPress={this.props.trigger1}>
                    <Image source={this.props.picture}/>
                </TouchableOpacity>
                :
                <TouchableOpacity onPress={this.props.trigger2}>
                    <Image source={this.props.picture}/>
                </TouchableOpacity>
                }
            </View>
        );
    }
}