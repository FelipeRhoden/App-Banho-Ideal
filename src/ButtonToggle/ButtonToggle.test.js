import React from 'react';
import renderer from "react-test-renderer";

import ButtonToggle from './';

describe("<ButtonToggle />", function(){

    it('Deveria ter 3 props (trigger1, trigger2, Picture)', ()=>{
        const Button = new ButtonToggle();

        expect(Button.props).toEqual(expect.objectContaining({
            trigger1: expect.anything(),
            trigger2: expect.anything(),
            picture: expect.anything(),
        }));
    });

    it('Deveria renderizar corretamente', ()=>{
        const menu = renderer.create(<ButtonToggle />).toJSON();

        expect(menu).toMatchSnapshot();
    });
    
})