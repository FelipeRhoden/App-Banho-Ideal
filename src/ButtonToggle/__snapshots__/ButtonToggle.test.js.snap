// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonToggle /> Deveria renderizar corretamente 1`] = `
<View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image />
  </View>
</View>
`;
